lib/m2ts/caption-stream.js

```diff
@@ -515,6 +515,14 @@ Cea608Stream.prototype.flushDisplayed = function(pts) {
   var content = this.displayed_
     // remove spaces from the start and end of the string
     .map(function(row) {
+      if (row === undefined || row === null) {
+        // Ordinarily, this shouldn't happen. However, caption
+        // parsing errors should not throw exceptions and
+        // break playback.
+        // eslint-disable-next-line no-console
+        console.log('Skipping malformed caption.');
```

Comment:
I had this as a `console.error` before, but noticed other streams like the MetadataStream had `console.log`s for malformed data. I could convert all of these in another PR if that makes sense, though I would assume an event would not be "loud" enough to get an implementor's attention

Reply:
It may also make sense to do `console.error` here, then convert the others in the future. I think we have to error to get the implementor's attention. That or if we want to trigger events instead, we can do that and log at a higher levels with the videojs log.

Reply:
I'm not sure this really indicates a malformed caption, though, so much as a potential problem in our state machine. Which is exactly what was causing the `undefined` here -- we weren't doing proper bounds checking.

A malformed caption would cause issues elsewhere.

Reply:
I suppose it's "malformed" in the sense that setting the base row to not match the number of roll up rows is incorrect. The language could probably be improved though.